### PR TITLE
Feature/healthcheck

### PR DESCRIPTION
**Pull request by swilson-mojhub**
Code changes to add backend and frontend endpoints

### CODE REVIEW

File: moj-be/modules/custom/moj_healthcheck/src/Controller/HealthcheckApiController.php [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    }
+
+		public function setResponse()
+    {
+        return $this->databaseSimpleTest() ? time() : 'FALSE';
+    }
+
+		public function databaseSimpleTest()
+    {
+        $connection = \Drupal::database();
```
Comment (glenjamin) [inline moj-be/modules/custom/moj_healthcheck/src/Controller/HealthcheckApiController.php]: what's the timeout set to for this? Ideally it should be <1 second.

File: moj-be/modules/custom/moj_healthcheck/src/Controller/HealthcheckApiController.php [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+use Drupal\Core\Database;
+
+class HealthcheckApiController extends ControllerBase
+{
+    public $response;
+
+    public function healthcheckApiEndpoint()
+    {
+        $this->response = $this->setResponse();
+        return $this->response;
```
Comment (glenjamin) [inline moj-be/modules/custom/moj_healthcheck/src/Controller/HealthcheckApiController.php]: I'd be nice to make this a bit richer - include the application version number / git reference etc. Not essential though.

It should also return a 200 if OK and a 500 if not.

File: moj-fe/app/Http/Controllers/HealthController.php [hunk 3]
```diff
@@ -3,17 +3,88 @@
...
+        return $this->cmsHealth;
+    }
+
+    public function __construct()
+    {
+        $this->response = new Response();
+        $this->client = new Client(
+          [
+            'base_uri' => config('app.api_uri'),
+            'timeout' => config('app.timeout'),
```
Comment (glenjamin) [inline moj-fe/app/Http/Controllers/HealthController.php]: What's this one set to? might be worth overriding to 2 seconds regardless of what the global setting is

File: moj-fe/app/Http/Controllers/HealthController.php [hunk 4]
```diff
@@ -3,17 +3,88 @@
...
+          ]
+        );
+    }
+
+    public function frontEndCheckHealthEndpoint()
+    {
+        $this->setHttpResponse(200);
```
Comment (glenjamin) [inline moj-fe/app/Http/Controllers/HealthController.php]: if we want separate checks we can have `/frontend-health` and `/backend-health`.

But the main `/health` check should be a full-stack check of all the bits.

Example from nomis web https://notm-dev.hmpps.dsd.io/health
```
{
  "name": "elite2-web",
  "version": "2018-03-05.2261",
  "description": "Elite 2 Web",
  "api": {
    "status": "UP",
    "healthInfo": {
      "status": "UP",
      "version": "1.0.21.2224"
    },
    "diskSpace": {
      "status": "UP",
      "total": 10725883904,
      "free": 10512019456,
      "threshold": 10485760
    },
    "db": {
      "status": "UP",
      "database": "Oracle",
      "hello": "Hello"
    }
  }
}
```

File: moj-be/modules/custom/moj_healthcheck/src/Controller/HealthcheckApiController.php (hunk 1 above)
Comment (swilson-mojhub) [reply moj-be/modules/custom/moj_healthcheck/src/Controller/HealthcheckApiController.php]: I've set this to 30 seconds

File: moj-be/modules/custom/moj_healthcheck/src/Controller/HealthcheckApiController.php (hunk 2 above)
Comment (swilson-mojhub) [reply moj-be/modules/custom/moj_healthcheck/src/Controller/HealthcheckApiController.php]: The git thing is going to take a bit more work which I'll card up.. 
If the database is down then drupal will give a 500

File: moj-fe/app/Http/Controllers/HealthController.php (hunk 3 above)
Comment (swilson-mojhub) [reply moj-fe/app/Http/Controllers/HealthController.php]: I've hard coded the time to 30s.. it was at 60s

File: moj-fe/app/Http/Controllers/HealthController.php (hunk 4 above)
Comment (swilson-mojhub) [reply moj-fe/app/Http/Controllers/HealthController.php]: I've switched to a full stack endpoint and changed the JSON to use a similar format to the example you gave.

**Pull request merged**